src/app/utils/api/fetchMovies.ts

```diff
@@ -0,0 +1,7 @@
+import { Movie } from "@prisma/client";
+import { PaginationSchema } from "../paginationFilter";
+
+export const fetchMovies = async ({ limit, page }: PaginationSchema) => {
+  const result = await fetch(`/api/movies/all?limit=${limit}&page=${page}`);
+  return (await result.json()) as Movie[];
```

Comment:
I find this a little hard to read. I might have done this instead:

`const movies: Movie[] = await res.json()`
`return movies`

Let me know what you think :)

Reply:
Agreed, I've updated it to your suggestion :)